engine/processor/src/main/java/io/deephaven/processor/NamedObjectProcessor.java

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.processor;
+
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.qst.type.Type;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Immutable;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Immutable
+@BuildableStyle
+public abstract class NamedObjectProcessor<T> {
```

Comment:
It throws me off that that this "has an" `ObjectProcessor` rather than "is an" `ObjectProcessor`.